### PR TITLE
[JENKINS-51598] - Expose PermissionGroup IDs to API

### PR DESCRIPTION
**Pull request by oleg-nenashev**
See [JENKINS-51598](https://issues.jenkins-ci.org/browse/JENKINS-51598).

Currently hudson.security.PermissionGroup API offers only getName() method, but this name is localizable. In Configuration-as-Code plugin we would like to use PermissionGroup IDs in order to define permissions from YAML.

ttps://github.com/jenkinsci/configuration-as-code-plugin/blob/configuration-as-code-0.7-alpha/src/main/java/org/jenkinsci/plugins/casc/util/PermissionFinder.java#L52-L56

It was worked around in https://github.com/jenkinsci/configuration-as-code-plugin/pull/240 , but IMHO a Core API would be useful

### Proposed changelog entries

* RFE - Developer: `PermissionGroup`s now expose IDs to API
* ...

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@MadsNielsen @ndeloof @ewelinawilkosz @jenkinsci/code-reviewers

### CODE REVIEW

File: core/src/main/java/hudson/security/PermissionGroup.java [hunk 1]
```diff
@@ -51,19 +52,41 @@
...
+    private final String id;
+
     /**
      * Both creates a registers a new permission group.
      * @param owner sets {@link #owner}
      * @param title sets {@link #title}
      * @throws IllegalStateException if this group was already registered
      */
     public PermissionGroup(@Nonnull Class owner, Localizable title) throws IllegalStateException {
+        this(title.toString(Locale.ENGLISH), owner, title);
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/security/PermissionGroup.java]: I didn't deprecate the method though. Can do so

**Review by ndeloof — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
Also @reviewbybees I'd guess

File: core/src/main/java/hudson/security/PermissionGroup.java [hunk 2]
```diff
@@ -110,7 +133,7 @@ public int compareTo(PermissionGroup that) {
 
         // among the permissions of the same group, just sort by their names
         // so that the sort order is consistent regardless of classloading order.
-        return id().compareTo(that.id());
+        return getOwnerClassName().compareTo(that.getOwnerClassName());
```
Comment (daniel-beck) [inline core/src/main/java/hudson/security/PermissionGroup.java]: Why these changes from `id()` to `getOwnerClassName()`?

Comment (daniel-beck) [reply core/src/main/java/hudson/security/PermissionGroup.java]: Nevermind: `id()` used to be the owner class, and is now the English title. So this just retains previous behavior.

Comment (oleg-nenashev) [reply core/src/main/java/hudson/security/PermissionGroup.java]: Yes

**Comment by oleg-nenashev** [comment]
@reviewbybees done
